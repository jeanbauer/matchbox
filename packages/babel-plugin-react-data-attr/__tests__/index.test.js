import * as babel from 'babel-core';
import babelPluginReactDataAttr from '../src';

function transform(code, pluginOptions, transformOptions) {
  return babel.transform(code, {
    babelrc: false,
    plugins: [
      [babelPluginReactDataAttr, pluginOptions]
    ],
    parserOpts: {
      plugins: ['jsx'],
      presets: ['@babel/preset-react']
    },
    ...transformOptions
  }).code.trim();
}

describe('babel-plugin-react-data-attr', () => {

  it.only('should transform a returned html element', () => {
    console.log(transform(`
      function MyComponent() {
        return <div />;
      }
    `));
    // expect(transform(`
    //   function MyComponent() {
    //     return <div />;
    //   }
    // `)).toMatchSnapshot();
  });

  it('should not transform a returned react component', () => {
    expect(transform(`
      function MyComponent() {
        return <TestComponent />;
      }
    `)).toMatchSnapshot();
  });

  it('should transform the first html element', () => {
    expect(transform(`
      function MyComponent() {
        return (
          <TestComponent>
            <div></div>
          </TestComponent>
        );
      }
    `)).toMatchSnapshot();
  });

});
