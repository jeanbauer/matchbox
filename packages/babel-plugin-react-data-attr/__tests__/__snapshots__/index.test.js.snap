// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-react-data-attr should not transform a returned react component 1`] = `
"function MyComponent() {
  return <TestComponent />;
}"
`;

exports[`babel-plugin-react-data-attr should transform a returned html element 1`] = `
"function MyComponent() {
  return <div />;
}"
`;

exports[`babel-plugin-react-data-attr should transform the first html element 1`] = `
"function MyComponent() {
  return <TestComponent>
            <div></div>
          </TestComponent>;
}"
`;
